pkg/R/types.R

```diff
@@ -0,0 +1,47 @@
+# Utility functions for handling Spark DataTypes.
+
+# Handler for StructType
+structType <- function(st) {
+  obj <- structure(new.env(parent = emptyenv()), class = "structType")
+  obj$jobj <- st
+  obj$fields <- lapply(SparkR:::callJMethod(st, "fields"), structField)
+  obj
+}
+
+#' Print a Spark StructType.
+#'
+#' This function prints the contents of a StructType returned from the
+#' SparkR JVM backend.
+#'
+#' @param x A StructType object
+#' @param ... further arguments passed to or from other methods
+print.structType <- function(x, ...) {
+  fieldsList <- lapply(x$fields, function(i) i$print)
+  print(fieldsList)
+}
+
+# Handler for StructField
+structField <- function(sf) {
+  obj <- structure(new.env(parent = emptyenv()), class = "structField")
+  obj$jobj <- sf
+  obj$name <- SparkR:::callJMethod(sf, "name")
+  obj$dataType <- SparkR:::callJMethod(sf, "dataType")
+  obj$dataType.toString <- SparkR:::callJMethod(obj$dataType, "toString")
+  obj$dataType.simpleString <- SparkR:::callJMethod(obj$dataType, "simpleString")
+  obj$nullable <- SparkR:::callJMethod(sf, "nullable")
+  obj$print <- paste("StructField(", 
```

Comment:
The problem is that since you can't string multiple method calls together in `callJMethod`, there isn't a good way to access the different values without having to just put more `callJMethod` calls in other functions. I think @shivaram's suggestion about making all of these lazy will allow us to have all of these attributes without it being too costly since none of them will actually execute when the `structField` is created.